### PR TITLE
[Frost Mage] Ice Lance Utilization

### PR DESCRIPTION
**Pull request by Sharrq**
Shows the % of Ice Lances that were shattered (With Fingers of Frost or Winter's Chill). Suggestions for not casting Ice Lance without procs unless you are moving and cant cast anything else instead.

### CODE REVIEW

File: src/Parser/Mage/Frost/Modules/Features/IceLance.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		}
+	}
+
+	on_byPlayer_damage(event) {
+		const spellId = event.ability.guid;
+		if (spellId !== SPELLS.ICE_LANCE_DAMAGE.id || event.targetID !== this.iceLanceTargetID) {
+			return;
+		}
+		const enemy = this.enemies.getEntity(event);
+		if (!enemy.hasBuff(SPELLS.WINTERS_CHILL.id) && this.hadFingersProc === 0) {
```
Comment (kfinch) [inline src/Parser/Mage/Frost/Modules/Features/IceLance.js]: Winter's Chill isn't the only debuff on target that can cause a Shatter. Also need to check for Water Elemental Freeze, Frost Nova, Ice Nova, Ring of Frost, and Glacial Spike (maybe more I'm not thinking of?). Probably should have an array of freeze effects to check against.

File: src/Parser/Mage/Frost/Modules/Features/IceLance.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+			this.nonShatteredCasts += 1;
+		}
+	}
+
+	suggestions(when) {
+		const utilization = 1 - (this.nonShatteredCasts / this.abilityTracker.getAbility(SPELLS.ICE_LANCE_CAST.id).casts);
+		const nonShatteredPercent = (this.nonShatteredCasts / this.abilityTracker.getAbility(SPELLS.ICE_LANCE_CAST.id).casts);
+		when(utilization).isLessThan(0.9)
+			.addSuggestion((suggest, actual, recommended) => {
+				return suggest(<span>You casted <SpellLink id={SPELLS.ICE_LANCE_CAST.id} /> {this.nonShatteredCasts} times ({formatPercentage(nonShatteredPercent)}%) with no procs. Make sure that you are only casting Ice Lance when the target has <SpellLink id={SPELLS.WINTERS_CHILL.id} />, if you have a <SpellLink id={SPELLS.FINGERS_OF_FROST.id} /> proc, or if you are moving and you cant cast anything else.</span>)
```
Comment (kfinch) [inline src/Parser/Mage/Frost/Modules/Features/IceLance.js]: Should be 'You cast', not 'You casted'.

'without Procs' is vague, say 'without Shatter'.

Update list of ways to shatter from "if target has Winter's Chill" to "if target has Winter's Chill or another Freeze effect"

File: src/Parser/Mage/Frost/Modules/Features/IceLance.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+			});
+	}
+
+	statistic() {
+		const utilization = 1 - (this.nonShatteredCasts / this.abilityTracker.getAbility(SPELLS.ICE_LANCE_CAST.id).casts);
+		return(
+			<StatisticBox
+				icon={<SpellIcon id={SPELLS.ICE_LANCE_CAST.id} />}
+				value={`${formatPercentage(utilization)}%`}
+				label='Ice Lance Utilization'
```
Comment (kfinch) [inline src/Parser/Mage/Frost/Modules/Features/IceLance.js]: Utilization is vague when it doesn't need to be, say "Ice Lances Shattered"

File: src/Parser/Mage/Frost/Modules/Features/IceLance.js (hunk 1 above)
Comment (Sharrq) [reply src/Parser/Mage/Frost/Modules/Features/IceLance.js]: I have all the IDs for these debuffs now, and i made the array ... but im not sure how to incorporate the array into hasBuff().

Comment (Sharrq) [reply src/Parser/Mage/Frost/Modules/Features/IceLance.js]: nvm, i think i found the answer in the warlock modules :P

**Comment by Sharrq** [comment]
I think i might have screwed something up. I clicked the Resolve Conflicts button and used the web editor that came up to fix the change log, and it looks like it made more changes than just the changelog. Dont know if thats what it was supposed to do, or if i messed something up ... apologies in advance if i made a dumb mistake.

**Comment by kfinch** [comment]
It's probably because you committed the rest of the merge from master along with the conflict resolution, which is expected and fine.

**Pull request merged**